src/addEffect.coffee

```diff
@@ -1,10 +1,42 @@
 import {useEffect} from 'react'
+import {isArray, get, shallowEqualArray} from './util/helpers'
+import usePrevious from './util/usePrevious'
 
-addEffect = (callback, changeProps) -> (props) ->
-  useEffect callback(props),
+isSimpleDependenciesArray = (dependencies) ->
+  return no unless isArray dependencies
+  for element in dependencies
+    if element.indexOf('.') > -1
+      return no
+  yes
+
+addEffect = (callback, dependencies) -> (props) ->
+  # eslint-disable-next-line coffee/no-negated-condition
+  if not dependencies?
+    useEffect callback props
+  else if isSimpleDependenciesArray dependencies
     # TODO: throw nice error if changeProps isn't array/iterable or any changeProp isn't a string?
-    if changeProps?
-      props[changeProp] for changeProp in changeProps
+    useEffect(
+      callback props
+      (props[dependencyPropName] for dependencyPropName in dependencies)
+    )
+  else
```

Comment:
Is the check for `isSimpleDependenciesArray` for efficiency, or would using the `else` branch in the simple case not actually work?

Reply:
I figured it made sense to continue to pass the simple case through to the underlying `useEffect()` dependencies argument

It presumably would behave the same way if we handled the dependency checking ourselves for that case as well, but may be more efficient to delegate to the underlying `useEffect()` dependency checking when we can

Speaking of efficiency though, I could move the `isSimpleDependenciesArray()` check to outside the inner `(props) ->` function (so that it doesn't get recomputed every render)

I think I'll go ahead and merge your `addLayoutEffect()` PR down into this branch/PR so that I can make that change in the shared helper rather than you having to re-rebase/manually recreate that change there